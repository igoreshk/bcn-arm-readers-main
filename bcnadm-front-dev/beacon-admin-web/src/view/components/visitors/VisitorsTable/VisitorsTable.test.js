import React from 'react';
import { shallow } from 'enzyme';

import { VisitorsTable } from './index';

const props = {
  translate: jest.fn(),
  showLoadingScreen: jest.fn(),
  hideLoadingScreen: jest.fn(),
  history: {},
  updated: false
};

const wrapper = shallow(<VisitorsTable {...props} />);

describe('VisitorsTable', () => {
  const loadData = jest.spyOn(VisitorsTable.prototype, 'loadData');
  const showLoadingScreen = jest.spyOn(wrapper.instance().props, 'showLoadingScreen');

  it('Checks that props are passed properly', () => expect(wrapper.instance().props).toEqual(props));

  it('Check showLoadingScreen was called from componentWillMount and componentDidUpdate life hooks', () => {
    wrapper.setProps({ updated: true });
    expect(showLoadingScreen).toHaveBeenCalledTimes(2);
  });

  it('Check loadData was called from componentDidUpdate life hook', () => expect(loadData).toHaveBeenCalled());

  it('Check matching snapshot', () => expect(wrapper).toMatchSnapshot());
});
