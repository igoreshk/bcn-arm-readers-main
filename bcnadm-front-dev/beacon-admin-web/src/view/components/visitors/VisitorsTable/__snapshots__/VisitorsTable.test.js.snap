// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisitorsTable Check matching snapshot 1`] = `
<WithStyles(ForwardRef(Table))
  className="visitorsTable"
>
  <WithStyles(ForwardRef(TableHead))>
    <VisitorsTableHeader
      sortedBy={null}
      toggleFilter={[Function]}
      toggleSortVisitorsByField={[Function]}
      translate={[MockFunction]}
    />
  </WithStyles(ForwardRef(TableHead))>
  <WithStyles(ForwardRef(TableBody)) />
</WithStyles(ForwardRef(Table))>
`;
